From: Jakub Adam <jakub.adam@ktknet.cz>
Date: Tue, 8 Nov 2011 22:54:49 +0100
Subject: debian-custom-build

- disable maven-source-plugin (we don't need jar archive with sources)
- disable findbugs-maven-plugin (not in Debian)
- disable maven-pmd-plugin (not in Debian)
- do not build org.eclipse.jgit.http.test (requires newer Jetty than present in Debian)
- do not build org.eclipse.jgit.storage.dht (requires packaging bug in libprotobuf-java fixed)
- add org.eclipse.jgit.pgm to jgit console classpath
- don't build jgit console all-in-one executable
---
 .../pom.xml                                        |    4 +--
 org.eclipse.jgit.http.server/pom.xml               |    4 +--
 org.eclipse.jgit.pgm/jgit.sh                       |    2 +-
 org.eclipse.jgit.pgm/pom.xml                       |    6 ++--
 org.eclipse.jgit.storage.dht/pom.xml               |   12 +++----
 org.eclipse.jgit/pom.xml                           |    8 ++---
 pom.xml                                            |   33 ++++++++++----------
 7 files changed, 35 insertions(+), 34 deletions(-)

diff --git a/org.eclipse.jgit.generated.storage.dht.proto/pom.xml b/org.eclipse.jgit.generated.storage.dht.proto/pom.xml
index 746d2d6..5f983f9 100644
--- a/org.eclipse.jgit.generated.storage.dht.proto/pom.xml
+++ b/org.eclipse.jgit.generated.storage.dht.proto/pom.xml
@@ -88,7 +88,7 @@
     </resources>
 
     <plugins>
-      <plugin>
+      <!--plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-source-plugin</artifactId>
         <inherited>true</inherited>
@@ -106,7 +106,7 @@
             </configuration>
           </execution>
         </executions>
-      </plugin>
+      </plugin-->
 
       <plugin>
         <artifactId>maven-jar-plugin</artifactId>
diff --git a/org.eclipse.jgit.http.server/pom.xml b/org.eclipse.jgit.http.server/pom.xml
index ede436a..fbc1d2a 100644
--- a/org.eclipse.jgit.http.server/pom.xml
+++ b/org.eclipse.jgit.http.server/pom.xml
@@ -97,7 +97,7 @@
     </resources>
 
     <plugins>
-      <plugin>
+      <!--plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-source-plugin</artifactId>
         <inherited>true</inherited>
@@ -115,7 +115,7 @@
             </configuration>
           </execution>
         </executions>
-      </plugin>
+      </plugin-->
 
       <plugin>
         <artifactId>maven-jar-plugin</artifactId>
diff --git a/org.eclipse.jgit.pgm/jgit.sh b/org.eclipse.jgit.pgm/jgit.sh
index 6b9ce40..3700d4f 100644
--- a/org.eclipse.jgit.pgm/jgit.sh
+++ b/org.eclipse.jgit.pgm/jgit.sh
@@ -86,7 +86,7 @@ Darwin)
 	;;
 esac
 
-CLASSPATH="$cp"
+CLASSPATH="$cp:/usr/share/java/org.eclipse.jgit.pgm.jar"
 export CLASSPATH
 
 java=java
diff --git a/org.eclipse.jgit.pgm/pom.xml b/org.eclipse.jgit.pgm/pom.xml
index 2c598b2..05c6316 100644
--- a/org.eclipse.jgit.pgm/pom.xml
+++ b/org.eclipse.jgit.pgm/pom.xml
@@ -105,7 +105,7 @@
       </dependencies>
     </profile>
   </profiles>
-
+	
   <build>
     <sourceDirectory>src/</sourceDirectory>
 
@@ -124,7 +124,7 @@
     </resources>
 
     <plugins>
-      <plugin>
+      <!--plugin>
         <artifactId>maven-jar-plugin</artifactId>
         <configuration>
           <archive>
@@ -237,7 +237,7 @@
             </configuration>
           </execution>
         </executions>
-      </plugin>
+      </plugin-->
     </plugins>
   </build>
 </project>
diff --git a/org.eclipse.jgit.storage.dht/pom.xml b/org.eclipse.jgit.storage.dht/pom.xml
index b7f81ad..da1d0a3 100644
--- a/org.eclipse.jgit.storage.dht/pom.xml
+++ b/org.eclipse.jgit.storage.dht/pom.xml
@@ -95,7 +95,7 @@
     </resources>
 
     <plugins>
-      <plugin>
+      <!--plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-source-plugin</artifactId>
         <inherited>true</inherited>
@@ -113,7 +113,7 @@
             </configuration>
           </execution>
         </executions>
-      </plugin>
+      </plugin-->
 
       <plugin>
         <artifactId>maven-jar-plugin</artifactId>
@@ -123,7 +123,7 @@
           </archive>
         </configuration>
       </plugin>
-         <plugin>
+         <!--plugin>
             <groupId>org.codehaus.mojo</groupId>
             <artifactId>findbugs-maven-plugin</artifactId>
             <configuration>
@@ -137,9 +137,9 @@
                   </goals>
                </execution>
             </executions>
-         </plugin>
+         </plugin-->
 
-         <plugin>
+         <!--plugin>
             <groupId>org.apache.maven.plugins</groupId>
             <artifactId>maven-pmd-plugin</artifactId>
             <configuration>
@@ -156,7 +156,7 @@
                   </goals>
                </execution>
             </executions>
-         </plugin>
+         </plugin-->
 
          <plugin>
             <groupId>org.codehaus.mojo</groupId>
diff --git a/org.eclipse.jgit/pom.xml b/org.eclipse.jgit/pom.xml
index edd522b..1869fad 100644
--- a/org.eclipse.jgit/pom.xml
+++ b/org.eclipse.jgit/pom.xml
@@ -115,7 +115,7 @@
         </executions>
       </plugin>
 
-      <plugin>
+      <!--plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-source-plugin</artifactId>
         <inherited>true</inherited>
@@ -133,7 +133,7 @@
             </configuration>
           </execution>
         </executions>
-      </plugin>
+      </plugin-->
 
       <plugin>
         <artifactId>maven-jar-plugin</artifactId>
@@ -152,13 +152,13 @@
 
     <pluginManagement>
       <plugins>
-        <plugin>
+        <!--plugin>
           <groupId>org.codehaus.mojo</groupId>
           <artifactId>findbugs-maven-plugin</artifactId>
           <configuration>
             <excludeFilterFile>findBugs/FindBugsExcludeFilter.xml</excludeFilterFile>
           </configuration>
-        </plugin>
+        </plugin-->
       </plugins>
     </pluginManagement>
   </build>
diff --git a/pom.xml b/pom.xml
index fdf5147..5a46f75 100644
--- a/pom.xml
+++ b/pom.xml
@@ -216,11 +216,11 @@
           <version>2.4.1</version>
         </plugin>
 
-        <plugin>
+        <!--plugin>
           <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-shade-plugin</artifactId>
           <version>1.5</version>
-        </plugin>
+        </plugin-->
 
         <plugin>
           <groupId>org.apache.maven.plugins</groupId>
@@ -234,11 +234,11 @@
           <version>2.3</version>
         </plugin>
 
-        <plugin>
+        <!--plugin>
           <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-source-plugin</artifactId>
           <version>2.1.2</version>
-        </plugin>
+        </plugin-->
 
         <plugin>
           <groupId>org.apache.maven.plugins</groupId>
@@ -258,7 +258,7 @@
           <version>1.5</version>
         </plugin>
 
-        <plugin>
+        <!--plugin>
           <groupId>org.codehaus.mojo</groupId>
           <artifactId>findbugs-maven-plugin</artifactId>
           <version>2.3.2</version>
@@ -273,9 +273,9 @@
               </goals>
             </execution>
           </executions>
-        </plugin>
+        </plugin-->
 
-        <plugin>
+        <!--plugin>
           <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-pmd-plugin</artifactId>
           <version>2.6</version>
@@ -293,7 +293,7 @@
               </goals>
             </execution>
           </executions>
-        </plugin>
+        </plugin-->
 
         <plugin>
           <groupId>org.codehaus.mojo</groupId>
@@ -363,6 +363,7 @@
           <links>
             <link>http://java.sun.com/j2se/1.5.0/docs/api</link>
           </links>
+          <detectOfflineLinks>false</detectOfflineLinks>
         </configuration>
         <executions>
           <execution>
@@ -373,7 +374,7 @@
           </execution>
         </executions>
       </plugin>
-      <plugin>
+      <!--plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-source-plugin</artifactId>
         <inherited>true</inherited>
@@ -386,7 +387,7 @@
             </goals>
           </execution>
         </executions>
-      </plugin>
+      </plugin-->
     </plugins>
   </build>
 
@@ -471,20 +472,20 @@
 
   <modules>
     <module>org.eclipse.jgit</module>
-	<module>org.eclipse.jgit.generated.storage.dht.proto</module>
-    <module>org.eclipse.jgit.storage.dht</module>
+	<!--module>org.eclipse.jgit.generated.storage.dht.proto</module-->
+    <!--module>org.eclipse.jgit.storage.dht</module-->
     <module>org.eclipse.jgit.ant</module>
     <module>org.eclipse.jgit.ui</module>
     <module>org.eclipse.jgit.http.server</module>
     <module>org.eclipse.jgit.iplog</module>
     <module>org.eclipse.jgit.pgm</module>
     <module>org.eclipse.jgit.junit</module>
-    <module>org.eclipse.jgit.junit.http</module>
+    <!--module>org.eclipse.jgit.junit.http</module-->
 
-    <module>org.eclipse.jgit.test</module>
+    <!--module>org.eclipse.jgit.test</module-->
     <module>org.eclipse.jgit.ant.test</module>
-    <module>org.eclipse.jgit.http.test</module>
-    <module>org.eclipse.jgit.storage.dht.test</module>
+    <!--module>org.eclipse.jgit.http.test</module-->
+    <!--module>org.eclipse.jgit.storage.dht.test</module-->
     <module>org.eclipse.jgit.pgm.test</module>
   </modules>
 
